### PR TITLE
DR | Fix verify modal design 

### PR DESCRIPTION
**Pull request by rocio-desantiago**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Moved component "NeedsToVerify" modal into a shared location for all 3 forms (HLR, NOD, SC) as part of initiative to refactor reusable and duplicate shared code (reduces future tech debt)
  > Updated verify modal based on design and content updates
- _(If bug, how to reproduce)_
  > n/a
- _(What is the solution, why is this the solution)_
  > The solution is to update the modal and place it in a new location for all 3 apps to use as it is the exact same design for all 3 forms
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Team 1 - Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
  > n/a

## Related issue(s)
- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#50637
- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#62362

## Testing done

- _Describe what the old behavior was prior to the change_
  > Content was different
- _Describe the steps required to verify your changes are working as expected_
  > Go to introduction page of each form
- _Describe the tests completed and the results_
  > Unit tests and locally

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Notice of Disagreement  | modalBefore | NOD_verifyModal |
| Supplemental Claims | modalBefore | SuppClaim_verifyModal |
| Higher Lever Reviews | modalBefore | HLR_verifyModal |

## What areas of the site does it impact?

  > Decision Reviews | Appeals (HLR, NOD, SC)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/shared/components/NeedsToVerify.jsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const NeedsToVerify = ({ pathname }) => {
           href={`/verify?next=${pathname}`}
           className="verify-link vads-c-action-link--green"
         >
-          Verify your identity
+          Verify your identity to start your request
```
Comment (Mottie) [inline src/applications/appeals/shared/components/NeedsToVerify.jsx:37]: We have:
- "Request a Board Appeal"
- "Request a Higher-Level Review"
- "File a Supplemental Claim"

I don't know if "request" works for Supplemental Claim? @eileen-coforma what do you think?

Comment (eileen-coforma) [reply src/applications/appeals/shared/components/NeedsToVerify.jsx:37]: What's the context? On va.gov most of the references towards Supplemental Claims use "File a Supplemental Claim". It's safe to stick with that.

Comment (rocio-desantiago) [reply src/applications/appeals/shared/components/NeedsToVerify.jsx:37]: For Supplemental Claims, do we want to update that action link to say `Verify your identity to file your claim`?

Comment (eileen-coforma) [reply src/applications/appeals/shared/components/NeedsToVerify.jsx:37]: I think for this, it's OK to say `Verify your identity to start your request` Thanks for the screenshot and extra context.

**Review by Mottie — APPROVED** [review]

**Pull request merged**